Kconfig.notecard

```diff
@@ -7,4 +7,8 @@ if BLUES_NOTECARD
 
 rsource "notecard/Kconfig"
 
+config REQUIRES_STD_C11
+	bool "Requires C11 for note-c library"
+    default y
```

Comment:
Why is this an option, if it can only ever be set to `Y`?

Reply:
This Kconfig option is not created here, but rather overriden its default value: other pieces of software can set a higher required version. So as opposed to forcing it to C11, the outcome would be whichever version satisfies all requirements.

Here's the definition for this and the other STD C options: https://github.com/zephyrproject-rtos/zephyr/blob/e4164963f22af072de43a6c56cbf0ad9f323ff1b/Kconfig.zephyr#L371-L392

It may be more clear to instead express it as a selelcted option:

```
menuconfig BLUES_NOTECARD
	...
	select REQUIRES_STD_C11
```

Does that make sense?